packages/react-instantsearch-hooks-dom/src/ui/RefinementList.tsx

```diff
@@ -0,0 +1,74 @@
+import { getHighlightedParts, unescape } from 'instantsearch.js/es/lib/utils';
+import React from 'react';
+
+import { Highlight } from './Highlight';
+import { cx } from './lib/cx';
+
+import type { RefinementListItem } from 'instantsearch.js/es/connectors/refinement-list/connectRefinementList';
+
+export type RefinementListProps = React.HTMLAttributes<HTMLDivElement> & {
+  items: RefinementListItem[];
+  onRefine(item: RefinementListItem): void;
+  searchBox?: React.ReactNode;
+  noResults?: React.ReactNode;
+  showMoreButton?: React.ReactNode;
+};
+
+export function RefinementList({
+  items,
+  onRefine,
+  searchBox,
+  noResults,
+  showMoreButton,
+  ...props
+}: RefinementListProps) {
+  return (
+    <div {...props} className={cx('ais-RefinementList', props.className)}>
+      {searchBox && (
+        <div className="ais-RefinementList-searchBox">{searchBox}</div>
+      )}
+      {noResults ? (
+        <div className="ais-RefinementList-noResults">{noResults}</div>
+      ) : (
+        <ul className="ais-RefinementList-list">
+          {items.map((item) => (
+            <li
+              key={item.value}
+              className={cx(
+                'ais-RefinementList-item',
+                item.isRefined && 'ais-RefinementList-item--selected'
+              )}
+            >
+              <label className="ais-RefinementList-label">
+                <input
+                  checked={item.isRefined}
+                  className="ais-RefinementList-checkbox"
+                  type="checkbox"
+                  value={item.value}
+                  onChange={() => {
+                    onRefine(item);
+                  }}
+                />
+                <span className="ais-RefinementList-labelText">
+                  <Highlight
```

Comment:
This means that we're always rendering highlighting DOM for the RefinementList, even when it's not searchable or that there's no query. Can we change this to support our IS.css spec?

Reply:
Good catch! There's not much in the [spec](https://instantsearch-css.netlify.app/widgets/refinement-list/) regarding the highlighting of searchable refinements, so I looked at the InstantSearch.js output. The only difference is that InstantSearch.js doesn't add classes to the `<mark>` elements and doesn't add tags around non-highlighted parts.

I think we should go with the default behavior of the `<Highlight>` component here, as this is more consistent with the other highlighted strings in `<Hits>`. By default, highlighted parts will have the same styles instead of showing up with the system background color for `<mark>`.

See 43e9451a.